packages/typescript/src/index.js

```diff
@@ -1,7 +1,42 @@
-import { generateTsConfigIfMissing } from './config-generation'
-import { registerTsNode } from './ts-node'
+import chalk from 'chalk'
+import consola from 'consola'
+import env from 'std-env'
+import { prompt } from 'enquirer'
+import { existsSync, writeJSON } from 'fs-extra'
+import { register } from 'ts-node'
 
-export async function setup(rootDir) {
-  await generateTsConfigIfMissing(rootDir)
-  registerTsNode()
+async function generateTsConfig(tsConfigPath) {
+  const configToExtend = '@nuxt/typescript'
+  await writeJSON(tsConfigPath, {
+    extends: configToExtend,
+    compilerOptions: {
+      baseUrl: '.',
+      types: [
+        '@types/node',
+        '@nuxt/vue-app'
+      ]
+    }
+  }, { spaces: 2 })
+  consola.info(`Extending ${chalk.bold.blue(`node_modules/${configToExtend}/tsconfig.json`)}`)
+  consola.success(`Generated successfully at ${chalk.bold.green(tsConfigPath)}`)
+}
+
+export async function setup(tsConfigPath) {
+  if (!existsSync(tsConfigPath)) {
```

Comment:
Can we add a guard to prevent calling function multiple times?

Reply:
What could be the case when it's called multiple times ?

How do we implement such `guard` ? :)

Reply:
Like this:

https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/setup.js#L13

> What could be the case when it's called multiple times?

TS Node will be registered twice :)

Reply:
I was meaning "In which case `setup` will be called twice ?", but I'll put a guard anyway ;)

Reply:
Thanks for the code ref

Reply:
Programmatic usage or when forking workers. But generally my reason is ensurance :)

Reply:
If I define a `_setup` variable, can it conflicts with `_setup` of CLI ? Or is `let` make it defined at the file scope ?

Reply:
@kevinmarrec rollup takes care of renaming even if they are bundled. No worries about same variable name or scope.

Reply:
Perfect then, thanks for the info 💪